### ISSUE TITLE
Saving and Loading Models

### ISSUE THREAD

**Issue by cortner** [ACEsuit/ACEpotentials.jl#217]
As discussed with @CheukHinHoJerry, it seems that the enourmous effort of implementing (de-)serialization routines we have in ACE1 is not required at all now that we've adopted the Lux model. 

Saving a potential should simply involve: 
- recording the model parameters
- recording the means how the model was generated
- recording the versions of relevant packages.

Loading a potential: 
- double-check version compatibility
- load the parameters
- generate the potential again, set the loaded parameters 

@CheukHinHoJerry @wcwitt  -- what are your thoughts on this?

**Comment by wcwitt** [comment]
Sorry just realized I was tagged here. Where can I look to see how this would work?

**Comment by cortner** [comment]
There is no example implemented. I should post a demo somewhere.

**Comment by cortner** [comment]
... following up. The gist of the proposal is this. A "model serialization" consists of 
- a script to generate it  (with the JSON interface, it would simply be the dictionary itself)
- a parameter file
- the manifest of the julia environment with with the model was originally generated and fitted.

**Comment by CheukHinHoJerry** [comment]
Following discussion with @cortner and @iamveronika might be helpful to save the manifest with the potential too, possibly with this package https://julialang.github.io/TOML.jl/dev/.

**Comment by cortner** [comment]
**SAVING A POTENTIAL**
```
user specifies model via Dict / JSON 
perform the fit 
Create new dict: 
    - user spec of model  (and also fitting parameters as meta-data)
    - parameters (as vector or as a named-tuple / Dict) 
    - Dict containing the Manifest 
Save this super-dict as JSON file (with option to zip)
```

**LOADING A POTENTIAL**
```
load super-dict from JSON
(maybe check compatibility with current environment, write warning if it has changed)
try to generate the model again from the model parameters
set the parameters  => model.
```

Further options: 
- instructions how to resolve the stored manifest if loading fails 
- add test set, e.g. a subset of the training data and check that original fitted potential and loaded potential evaluate to the same energies, forces, virials.

**Comment by cortner** [comment]
This was addressed in a direct commit to main (by accident)
